Remove IAP code and usage

iamgabrielma opened the pull request:
Closes WOOMOB-1290

## Description
This PR gets rid of the In-App Purchases implementation, as well as partially of the usage of Plan/Domain upgrades through onboarding flow, since some bits of this flow was also using IAP internals. Confirmed with Kiwi that is no longer needed here: p1757921125873789-slack-C03L1NF1EA3, but still needs some extra task to remove the remains that are not coupled, which can be done separately.

There has been no IAP purchases (merchants were not able to purchase since the feature was disabled) or renewals (existing purchases could still be renewed via the app) since April 5th, 2024, so we should be good to remove it. Android counterpart was done in January: pdfdoF-6aJ-p2

## Testing information
- App should run normally
- CI should pass

iamgabrielma committed:
remove upgrades entry point

iamgabrielma committed:
remove upgrades-subscriptions views

iamgabrielma committed:
remove iap models

iamgabrielma committed:
remove IAP action dispatcher

iamgabrielma committed:
remove IAP remotes and tests

iamgabrielma committed:
periphery cleanup

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is larger than 300 lin...
  1 Message: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma commented on Modules/Sources/Networking/Remote/PaymentRemote.swift:
```diff
@@ -28,6 +28,7 @@ public protocol PaymentRemoteProtocol {
 
 /// WPCOM Payment Endpoints
 ///
+// periphery:ignore
```
Temporary periphery ignores so I don't have to update `PaymentRemote`, is possible that we can remove these as well as part of 12401-gh continuation, since are related to domain purchases.

iamgabrielma committed:
update note

iamgabrielma commented:
@itsmeichigo looping you in as well for awareness

iamgabrielma commented on Modules/Sources/Networking/Remote/FeatureFlagRemote.swift:
```diff
@@ -28,15 +28,12 @@ public class FeatureFlagRemote: Remote, FeatureFlagRemoteProtocol {
         switch rawValue {
         case "woo_notification_store_creation_ready":
             self = .storeCreationCompleteNotification
-        case "woo_hardcoded_plan_upgrade_details_milestone_1_are_accurate":
```
We'll also have to clean the flag from `https://public-api.wordpress.com/wpcom/v2/mobile/feature-flags`. I haven't tackled this as part of the PR.

iamgabrielma commented:
👋 @joshheald Feel free to ignore this review during HACK week and team meetup, I'll move the milestone before I forget.

iamgabrielma committed:
Merge branch 'trunk' into task/WOOMOB-1290-remove-IAP-usage

iamgabrielma committed:
Merge branch 'trunk' into task/WOOMOB-1290-remove-IAP-usage

# Conflicts:
#	WooCommerce/Classes/ViewRelated/Dashboard/Free Trial/StorePlanBannerPresenter.swift
#	WooCommerce/Classes/ViewRelated/Upgrades/CurrentPlanDetailsView.swift
#	WooCommerce/Classes/ViewRelated/Upgrades/UpgradeWaitingView.swift
#	WooCommerce/WooCommerce.xcodeproj/project.pbxproj

iamgabrielma commented:
Additionally there are 2 subscription groups (9 subscriptions total, production and test) that we can remove from ASC, as well as the 2 .storekit config files from the Woo target. I'll clean these after we've wrapped this one up.

iamgabrielma committed:
Merge branch 'trunk' into task/WOOMOB-1290-remove-IAP-usage

joshheald commented:
I've not forgotten about this, will get to it in the next day or two

joshheald commented:
> Additionally there are 2 subscription groups (9 subscriptions total, production and test) that we can remove from ASC, as well as the 2 .storekit config files from the Woo target. I'll clean these after we've wrapped this one up.

I don't _think_ you can ever remove approved subscriptions from ASC, just mark them as not available for sale, which should be done already

joshheald reviewed (approved):
Well done. Always good to see a summer's worth of work deleted 😂 

There may be some stuff you can remove from PaymentAction/Store... worth checking with Kiwi whether we actually do the check for whether it's a free trial or not and show publishing stuff as part of the onboarding. It smells like Woo Express code to me. Another PR might be best though

iamgabrielma committed:
Merge branch 'trunk' into task/WOOMOB-1290-remove-IAP-usage

# Conflicts:
#	WooCommerce/WooCommerce.xcodeproj/project.pbxproj

iamgabrielma commented:
Thanks for the review!

> Always good to see a summer's worth of work deleted 😂

😂 

> There may be some stuff you can remove from PaymentAction/Store... worth checking with Kiwi whether we actually do the check for whether it's a free trial or not and show publishing stuff as part of the onboarding. It smells like Woo Express code to me. Another PR might be best though

Yes, there are still bits of IAP/WooExpress/Domains here and there that wasn't coupled with the rest of IAP, I'll clean up the other bits on a separate PR so this one doesn't get even bigger.

iamgabrielma committed:
Merge branch 'trunk' into task/WOOMOB-1290-remove-IAP-usage

iamgabrielma committed:
remove references from deleted files from woo build phase

Does not compile after modularization due to error: Build input files cannot be found. 4 files still appear referenced in the woo target build phase despite being deleted, so I’ve removed them manually.
